### PR TITLE
Detect whether coin daemon is running before funding

### PR DESCRIPTION
**Pull request by Varunram**

### CODE REVIEW

File: wallit/init.go [hunk 1]
```diff
@@ -107,8 +109,27 @@ func NewWallit(
 
 	// deal with incoming height
 	go w.HeightHandler(incomingBlockheight)
-
-	return &w
+	if !hookFail {
+		switch p.Name {
```
Comment (adiabat) [inline wallit/init.go]: I think you can do something like 
```
return &w, p.HDCoinType, nil
```
instead

Comment (Varunram) [reply wallit/init.go]: Yeah, was about to get to this, got stuck at the jenkins file thing. updating, thanks!

**Comment by metalicjames** [comment]
retest this please

**Comment by Varunram** [comment]
Rebased

**Pull request merged**